src/renderer/components/subscriptions-videos/subscriptions-videos.js

```diff
@@ -0,0 +1,492 @@
+          return (!item.liveNow && !item.isUpcoming)
+        })
+      }
+      if (this.hideUpcomingPremieres) {
+        videoList = videoList.filter(item => {
+          if (item.isRSS) {
+            // viewCount is our only method of detecting premieres in RSS
+            // data without sending an additional request.
+            // If we ever get a better flag, use it here instead.
+            return item.viewCount !== '0'
+          }
+          // Observed for premieres in Local API Subscriptions.
+          return item.premiereDate == null
+        })
+      }
+
+      this.videoList = videoList.filter((video) => {
+        if (this.hideWatchedSubs) {
+          const historyIndex = this.historyCache.findIndex((x) => {
+            return x.videoId === video.videoId
+          })
+
+          return historyIndex === -1
+        } else {
+          return true
+        }
+      })
+    },
+
+    maybeLoadVideosForSubscriptionsFromRemote: async function () {
+      if (this.fetchSubscriptionsAutomatically) {
+        // `this.isLoading = false` is called inside `loadVideosForSubscriptionsFromRemote` when needed
+        await this.loadVideosForSubscriptionsFromRemote()
+      } else {
+        this.videoList = []
+        this.attemptedFetch = false
+        this.isLoading = false
+      }
+    },
+
+    getChannelVideosLocalScraper: async function (channel, failedAttempts = 0) {
+      try {
+        const videos = await getLocalChannelVideos(channel.id)
+
+        if (videos === null) {
+          this.errorChannels.push(channel)
+          return []
+        }
+
+        videos.map(video => {
+          if (video.liveNow) {
+            video.publishedDate = new Date().getTime()
+          } else if (video.isUpcoming) {
+            video.publishedDate = video.premiereDate
+          } else {
+            video.publishedDate = calculatePublishedDate(video.publishedText)
+          }
+          return video
+        })
+
+        return videos
+      } catch (err) {
+        console.error(err)
+        const errorMessage = this.$t('Local API Error (Click to copy)')
+        showToast(`${errorMessage}: ${err}`, 10000, () => {
+          copyToClipboard(err)
+        })
+        switch (failedAttempts) {
+          case 0:
+            return await this.getChannelVideosLocalRSS(channel, failedAttempts + 1)
+          case 1:
+            if (this.backendFallback) {
+              showToast(this.$t('Falling back to Invidious API'))
+              return await this.getChannelVideosInvidiousScraper(channel, failedAttempts + 1)
+            } else {
+              return []
+            }
+          case 2:
+            return await this.getChannelVideosLocalRSS(channel, failedAttempts + 1)
+          default:
+            return []
+        }
+      }
+    },
+
+    getChannelVideosLocalRSS: async function (channel, failedAttempts = 0) {
+      const feedUrl = `https://www.youtube.com/feeds/videos.xml?channel_id=${channel.id}`
```

Comment:
We could use the UULF playlist here instead of channel rss (it'll exclude live + shorts).

I'm also wondering if we should include an all tab?(This can be done in a different PR if needed)

Reply:
All tab with refresh button = easier to get rate limited
No for me :S

Reply:
I guess the all tab may only work if it uses rss in that case and should probably be added in a future PR

Reply:
I don't know if RSS feed includes all types of videos (videos + live + short)
But I would consider RSS feed as a different type of "stream" and put into another tab / view
That's what I did in my branch of live implementation (but less complete than this PR)